tket2/src/ops.rs

```diff
@@ -226,29 +229,32 @@ impl TryFrom<&OpType> for Tk2Op {
     type Error = NotTk2Op;
 
     fn try_from(op: &OpType) -> Result<Self, Self::Error> {
-        let OpType::CustomOp(custom_op) = op else {
-            return Err(NotTk2Op);
-        };
-
-        match custom_op {
-            CustomOp::Extension(ext) => Tk2Op::from_extension_op(ext),
-            CustomOp::Opaque(opaque) => {
-                if opaque.extension() != &EXTENSION_ID {
-                    return Err(NotTk2Op);
-                }
-                try_from_name(opaque.name())
-            }
-        }
-        .map_err(|_| NotTk2Op)
+        optype_to_tk2op(op)
     }
 }
 
 impl TryFrom<OpType> for Tk2Op {
     type Error = NotTk2Op;
 
     fn try_from(op: OpType) -> Result<Self, Self::Error> {
-        Self::try_from(&op)
+        optype_to_tk2op(&op)
+    }
+}
+
+// Internal implementation for `TryFrom<Optype> for Tk2Op` that doesn't copy the `OpType` when it errors.
```

Comment:
Ah, the original function that returned an `Option` was meant as an optimisation to avoid cloning the optype on the error path of `TryFrom<OpType>`. I guess it's too much of a micro optimisation, so I'll just leave the `TryFrom<&OpType` impl as we had before.